frontend/apps/cloudstorage/webdavapi.lua

```diff
@@ -95,20 +98,27 @@ function WebDavApi:listFolder(address, user, pass, folder_path)
         return nil
     end
 
-    local res_data = table.concat(sink)
+    local res = table.concat(sink)
+    --[[
+    Apache Webserver WebDAV mod_dav module adds this fragment to the XML tags
+    It also uses upper casing for the XML tags, amongst other things, which we
```

Comment:
Hm? XML is case sensitive. Judging by a quick peek at https://tools.ietf.org/html/rfc4918 only `<D>` is correct. Pinging @yparitcher

Reply:
OK. Makes me further wonder what WebDAV server this code was originally written against.

Reply:
See https://github.com/koreader/koreader/commit/a3a17dbbeb7bcd94702c2f12acf1e09809340b50#diff-fff66d39dd02c73d9bc768e0e65b97eb

> Tested with:
NextCloud
HubZilla

Reply:
I think that in a XML tag with a namespace nickname/prefix like `<BlaH:response>`, the `response` part must be in the case specificed in the schema. The namespace nickname part `BlaH` can be in any case as long as it's the same that was specified in the XML document with `xmlns:BlaH="DAV:"` or `xmlns:BlaH="http://something"`
So, logically, you should either fetch the nickname/prefix specified to strip it from tags.
But may be in our context (we know it's a webdav response), just accept any nickname/prefix, not just `D` or `d` or `lp1`, and even the absence of any prefix (we should accept `<response>`) (but skip that absence case if it complicates thing and if all webdav server use a prefix).

Reply:
@poire-z 
Thanks, the part I was looking at actually didn't include a full XML document, though even if D were the default of course you could do xmlns:d="D:". But there's a complete sample included further down.

It's probably easier/more fruitful to toss in that real XML parser than to mess about with Lua patterns though.

Reply:
Agreed that we should use a real XML parser over Lua patterns. When I get a bit more time, I will test this against NextCloud and HubZilla (unless someone else gets to it before me).

Whoever implements the real XML parser should probably include some functional tests too, so we don't have to manually test every code change.

(I'm already using this on my kobo touch so I know it works.)

Reply:
@Frenzie I have no clue about XML :) if thats why you pinged me. did you intend for @y-muller who added the webDAV code?

Reply:
@yparitcher D'oh, must've misclicked. But thanks to @poire-z's observation we've already figured it out. :-)